File: 7-kyu/bus-mastering---who-is-the-most-prioritary--/c++/solution.cpp
using namespace std;

string arbitrate(const string& input, int n)
{
    string ret = "";
    bool found_master = false;
    for (const char &c : input) {
        if (c == '1' && found_master == false) {
            found_master = true;
            ret += "1";
        }
        else ret += "0";
    }
    return ret;
}

File: 7-kyu/bus-mastering---who-is-the-most-prioritary--/javascript/solution.js
const arbitrate = (input, n) => {
  for (let i = 0; i < input.length; i++) {
    if (input[i] === "1") {
      return "0".repeat(i) + "1" + "0".repeat(n - i - 1);
    }
  }
  return input;
};
